### PR TITLE
Move common commands and page-objects to base repo and re-structure folders

### PR DESCRIPTION
**Pull request by emswbandara**

### CODE REVIEW

File: src/api/commands/application-mgt/commands.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (NilukaSripalim) [inline src/api/commands/application-mgt/commands.js:2]: check the license header. shall we change with the current year?

Comment (aaujayasena) [inline src/api/commands/application-mgt/commands.js:2]: Shall we update the copyright year. please check on other files as well.

File: src/api/commands/application-mgt/oauth2.0/commands.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            'username': clientId,
+            'password': clientSecret,
+        },
+        headers:
+            {
+                'Content-Type': 'application/x-www-form-urlencoded'
+            },
+        body: regBody
+    });
+})
```
Comment (NilukaSripalim) [inline src/api/commands/application-mgt/oauth2.0/commands.js:36]: EOL

File: src/api/commands/common/commands.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+/**
+ * This method will base64 encode the given string
+ * @param  {} "Base64Encoding"
+ * @param  {} "string"
+ */
+Cypress.Commands.add("base64Encoding",(string)=>{
+    var encodedString = btoa(string);
+    console.log(encodedString); 
+    return encodedString;
+});
```
Comment (NilukaSripalim) [inline src/api/commands/common/commands.js:108]: EOL

File: src/ui/page-objects/console/application/application-attribute.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+     * Add application User Attributes
+     * @param  {} testData
+     */
+    static selectApplicationUserAttribute() {
+        cy.get(TAB_USER_ATTRIBUTE).click();
+
+    }
+}
+
+export default applicationAttributes;
```
Comment (NilukaSripalim) [inline src/ui/page-objects/console/application/application-attribute.js:25]: EOL

File: src/ui/page-objects/console/application/application-page.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
+
+/**
+ * Copyright 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (NilukaSripalim) [inline src/ui/page-objects/console/application/application-page.js:3]: check the license header. shall we change with the current year

File: src/ui/page-objects/myaccount/operation-page.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+const hdrPage = () => cy.get(LBL_PAGE_HEADER).invoke("text").then((text) => {
+
+    expect(text.trim()).equal(OPERATION_PAGE_TITLE);
+});
+
+module.exports = {
+
+    hdrPage
+};
```
Comment (aaujayasena) [inline src/ui/page-objects/myaccount/operation-page.js:21]: Shall we add  line here. Please check other files as well

File: src/ui/commands/common/commands.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+ * Copyright 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * This software is the property of WSO2 Inc. and its suppliers, if any.
+ * Dissemination of any information or reproduction of any material contained
+ * herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+ * You may not alter or remove any copyright or other notice from copies of this content."
+ */
+/// <reference types="cypress" />
+
+
```
Comment (NilukaSripalim) [inline src/ui/commands/common/commands.js:11]: Shall we remove the empty line

File: src/ui/commands/common/commands.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+ * This method is used to validate the URL is valid.
+ */
+Cypress.Commands.add("isUrlValid", (userInput) => {
+    var pattern = /(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?/;
+    if (pattern.test(userInput)) {
+        return true;
+    } 
+    return false;
+});
+
```
Comment (NilukaSripalim) [inline src/ui/commands/common/commands.js:53]: Shall we remove the empty line

File: src/ui/page-objects/console/application/application-attribute.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * This software is the property of WSO2 Inc. and its suppliers, if any.
+ * Dissemination of any information or reproduction of any material contained
+ * herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+ * You may not alter or remove any copyright or other notice from copies of this content."
+ */
+
```
Comment (NilukaSripalim) [inline src/ui/page-objects/console/application/application-attribute.js:9]: shall we add cypress reference "/// <reference types="cypress" />"

File: src/ui/page-objects/console/application/application-general.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
+
+/**
+ * Copyright 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * This software is the property of WSO2 Inc. and its suppliers, if any.
+ * Dissemination of any information or reproduction of any material contained
+ * herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+ * You may not alter or remove any copyright or other notice from copies of this content."
+ */
+
```
Comment (NilukaSripalim) [inline src/ui/page-objects/console/application/application-general.js:10]:
```
/// <reference types="cypress" />
```

File: src/ui/page-objects/console/application/application-page.js [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+ * Dissemination of any information or reproduction of any material contained
+ * herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+ * You may not alter or remove any copyright or other notice from copies of this content."
+ */
+
+import developSection from "../console/console-landing-page.js";
+import appManagment from "../application/application-page-object/application-common.js";
+import appProtocolMgt from "../application/application-page-object/application-protocol.js";
+
+class application {
```
Comment (brionmario) [inline src/ui/page-objects/console/application/application-page.js:15]:
```suggestion
class Application {
```

File: src/ui/page-objects/console/application/application-common.js [hunk 12]
```diff
@@ -0,0 +1,473 @@
+/**
+ * Copyright 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (ShanikaWickramasinghe) [inline src/ui/page-objects/console/application/application-common.js:2]: Check for license

File: src/ui/page-objects/console/application/application-common.js [hunk 13]
```diff
@@ -0,0 +1,473 @@
...
+
+    /**
+     * This method is used to click application edit button
+     */
+    static clickApplicationEditButton() {
+       
+        cy.wait(7000);
+        cy.get(EDIT_BUTTON).click({ force: true });
+    }    
+}
```
Comment (ShanikaWickramasinghe) [inline src/ui/page-objects/console/application/application-common.js:473]: EOL

File: src/ui/page-objects/console/application/application-general.js [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+class applicationGeneral {
+
+    // TODO: Add Application General page related constants and methods
+}
```
Comment (ShanikaWickramasinghe) [inline src/ui/page-objects/console/application/application-general.js:14]: EOL

File: src/ui/page-objects/console/application/application-general.js [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+/**
+ * Copyright 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * This software is the property of WSO2 Inc. and its suppliers, if any.
+ * Dissemination of any information or reproduction of any material contained
+ * herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+ * You may not alter or remove any copyright or other notice from copies of this content."
+ */
+
+class applicationGeneral {
```
Comment (ShanikaWickramasinghe) [inline src/ui/page-objects/console/application/application-general.js:11]: shall we use pascal scale when defining classs names which is JS standard

File: src/ui/page-objects/console/application/application-protocol.js [hunk 16]
```diff
@@ -0,0 +1,192 @@
...
+
+    /**
+     * Click Single Page Application Register button
+     */
+    static clickSinglePageApplicationRegisterBtn(){
+        
+        cy.get(BTN_SINGLE_APP_REGISTER).click();
+    }
+}
+export default applicationProtocol;
```
Comment (ShanikaWickramasinghe) [inline src/ui/page-objects/console/application/application-protocol.js:192]: EOL

File: src/ui/page-objects/console/identity-providers/identity-provider-page.js [hunk 17]
```diff
@@ -0,0 +1,14 @@
+/**
+ * Copyright 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * This software is the property of WSO2 Inc. and its suppliers, if any.
+ * Dissemination of any information or reproduction of any material contained
+ * herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+ * You may not alter or remove any copyright or other notice from copies of this content."
+ */
+
+ class identityProvider {
```
Comment (ShanikaWickramasinghe) [inline src/ui/page-objects/console/identity-providers/identity-provider-page.js:10]: shall we use pascal scale when defining classs names which is JS standard

File: src/ui/page-objects/myaccount/overview-page.js [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+const lnkSecurity = () => cy.contains(LNK_SECURITY);
+const lnkOperations = () => cy.contains(LNK_OPERATIONS);
+
+module.exports = {
+
+    lnkOperations,
+    lnkPersonalInfo,
+    lnkSecurity,
+    txtWelcome
+};
```
Comment (ShanikaWickramasinghe) [inline src/ui/page-objects/myaccount/overview-page.js:45]: EOL

File: src/ui/page-objects/myaccount/security-page.js [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+//Methods to get locators
+const hdrPage = () => cy.get(LBL_PAGE_HEADER).invoke("text").then((text) => {
+
+    expect(text.trim()).equal(SECURITY_PAGE_TITLE);
+});
+
+module.exports = {
+
+    hdrPage
+};
```
Comment (ShanikaWickramasinghe) [inline src/ui/page-objects/myaccount/security-page.js:25]: EOL

**Pull request closed without merging**